src/init.c

```diff
@@ -1535,6 +1537,35 @@ cmd_spec_header (const char *com, const char *val, void *place_ignored _GL_UNUSE
   return true;
 }
 
+static bool
+cmd_spec_resolve_host (const char *com _GL_UNUSED, const char *val, void *place_ignored _GL_UNUSED)
+{
+  //parse host:ip and put into dns cache
+
+  char* pos = strpbrk(val, ":");
+  bool result = false;
+  if (pos && pos != val && pos != val + strnlen(val, 256))
+  {
+    size_t len = strnlen(val, 256);
+    char* host = xmalloc(pos-val+1);
+    char* ip = xmalloc(len-(pos-val)+1);
+    strncpy(host, val, pos-val);
+    strncpy(ip, pos+1, val+len-pos);
+
+    result = resolve(host, ip);
```

Comment:
if resolve expected pointer to start and pointer after the end (stl like iterators), you would not need to touch allocator and copy data:
```
bool resolve(const char* host_start, const char* host_end, const char* ip_start, const char* ip_end);
resolve(val, pos, pos + 1,val + len);
```